src/components/data-management/SamplesTable.jsx

```diff
@@ -96,9 +96,16 @@ const SamplesTable = forwardRef((props, ref) => {
   const [tableColumns, setTableColumns] = useState(initialTableColumns);
 
   useEffect(() => {
-    if (activeExperiment.sampleIds.length > 0) {
+    const samplesLoaded = activeExperiment?.sampleIds.every((sampleId) => samples[sampleId]);
+
+    if (activeExperiment?.sampleIds.length > 0 && samplesLoaded) {
       // if there are samples - build the table columns
-      setSampleNames(new Set(activeExperiment.sampleIds.map((id) => samples[id]?.name.trim())));
+
+      const sanitizedSampleNames = new Set(
+        activeExperiment.sampleIds.map((id) => samples[id]?.name.trim()),
+      );
+
+      setSampleNames(sanitizedSampleNames);
```

Comment:
We don't load the samples immediately as soon as we start, so I added some checks that the samples are loaded first before going through with rendering